File: src/main/java/net/maple/handlers/net/PongHandler.java
package net.maple.handlers.net;

import client.Client;
import net.maple.handlers.PacketHandler;
import util.packet.PacketReader;

public class PongHandler extends PacketHandler {

    @Override
    public void handlePacket(PacketReader reader, Client c) {
        c.setLastPong(System.currentTimeMillis());
    }

    @Override
    public boolean validateState(Client c) {
        return true;
    }
}

File: src/main/java/net/maple/handlers/net/PongHandler.kt
package net.maple.handlers.net

import client.Client
import net.maple.handlers.PacketHandler
import util.packet.PacketReader

class PongHandler : PacketHandler {

    override fun handlePacket(reader: PacketReader, c: Client) {
        c.lastPong = System.currentTimeMillis()
    }

    override fun validateState(c: Client): Boolean {
        return true
    }
}